File: 0187) Repeated DNA Sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        int n = s.size();
        vector<string> ans;
        unordered_map<string, int> mp;
        if(n < 10)
        {
            return ans;
        }
        for(int i = 0; i <= n - 10; i++)
        {
            string check = s.substr(i, 10);
            mp[check]++;
        }
        for(auto it : mp)
        {
            if(it.second > 1)
            {
                ans.push_back(it.first);
            }
        }
        return ans;
    }
};

File: 0187) Repeated DNA Sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        int n = s.length();
        List<String> ans = new ArrayList<>();
        if(n < 10)
        {
            return ans;
        }
        HashMap<String, Integer> mp = new HashMap<>();
        for(int i = 0; i <= n-10; i++)
        {
            String check = s.substring(i, i+10);
            mp.put(check, mp.getOrDefault(check, 0) + 1);
        }
        for(HashMap.Entry<String, Integer> it : mp.entrySet())
        {
            if(it.getValue() > 1)
            {
                ans.add(it.getKey());
            }
        }
        return ans;
    }
}
